robotino_description/urdf/base.urdf.xacro

```diff
@@ -5,22 +5,86 @@
 
   <xacro:macro name="base" params="name">
   
-    <!--link name="${name}_footprint">
-   <visual>
+   <joint name="origin_joint" type="prismatic">
```

Comment:
@ipa-nhg : ok i see. so the overall picture. the problem was that the festo software automatically updates base_link. and the same position data (for the base links) was used for base_linkz which was directly attached to base_link. so every movement affected base_linkz twice. however, i also couldnt attach the arm to base_link because the 8-dof controller (uibk and bonn) requires the structure (jointx, jointy, jointz, arm joints) . this problem was solved to basically maintain 2 models of the robot (which is a really ugly solution - i was very much against that) which are now kept in sync. that are the different chains. they are both connected by the origin link (thats why the origin joint is there which always stays constant - if there is another way to connect two links that would also be fine for the functionality). the origin link is then connect by a static state publisher to odomp so that everything stays in sync. so basically, base_link and base_linkz are the same thing and will also be in the same position. the kinect etc is attached to base_linkz and not to base_link because they would stay still during planning and therefore the 8dof planner would always think there is collisions between the base and kinect etc (that is also the reason why i was against the third base_link... which federico suggested). so in general, new components should always be attached to base_linkz.

i can also sketch the clean solution that i would have had in mind: the main problem before was that festo is updating the base_link position. if that wouldnt be the case, base_link could just replace base_linkz in the model with the arm (the models without the arm could remain the same). then only the 8-dof controller would have to update base_link (which was base_linkz before) and everything would be fine again. the advantage would be that navigation would not have to care whether or not the arm is on the robot and everything would just be one clean chain. however, i decided to leave the project, so i cannot implement this. if you need further explanation to clear stuff up for whoever takes this over, we should talk soon